src/meta/src/manager/catalog/mod.rs

```diff
@@ -3420,8 +3420,6 @@ impl CatalogManager {
         database_core.ensure_database_id(table.database_id)?;
         database_core.ensure_schema_id(table.schema_id)?;
 
-        assert!(table.dependent_relations.is_empty());
```

Comment:
cc @yezizp2012 Do you think it's okay if we remove the assertion here? I can't remember why I included it. 😕

Reply:
I think it's okay since cdc table might dependents on a shared source? I guess it's to prevent misuse of alter mv or something like that. 🤔  Maybe we can add a table type assertion here.

Reply:
added a check